### ISSUE TITLE
Better Documentation of how to write a GPU Op

### ISSUE THREAD

**Issue by Sentient07** [Theano/Theano#5708]
This documentation [page](http://deeplearning.net/software/theano_versions/dev/extending/extending_theano_gpu.html) lacks a lot of information. 
It is not clear and isn't explained how `&ls` and `&gs` are set.  In some cases only single address is passed(Pooling Ops) and here there are two, ls and gs. Also, there is no specific reason why the first parameter is 1.
The `LID_0` and `LDIM_0` are also not explained and it is unclear why we increment with `LDIM_0`. 
This documentation page does not talk about the helper function that should be used to prepare outputs, `theano_prep_output`
In this example, the kernel (say `rankernel`) is called by `rankernel_call`. When we see examples of calling a kernel in Pooling operations, they're called using `rankernel_scall`. It is confusing when to use `call` and when to use `scall`. 

(There were a couple of more suggestions that I had for improving the doc. I had noted down elsewhere and am not able to reach to it)

**Comment by alkoa** [comment]
Did you find any good source of information to write a GPU Op. The documentation does not help at all.

**Comment by nouiz** [comment]
I don't think we will make more documentation, but if someone make a PR we will review and merge. Rapidly, ls and gs are some GPUs properties. "ls" is for "local" and "gs" is for "global". On cuda, ls correspond to "block dimensions" and gs to "grid dimensions". LID_0 is a macro that map to threadIdx.x in CUDA. LDIM_0 to blockDim.x. The _1 correspond to the ".y" version and so on.

theano_prep_output is an helper function that will allocate of the right size if the output isn't allocated or of the wrong size.

Thanks for your comment, if we end up updating this doc, we will check them again.
